flink-connector-aws/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/config/ConsumerConfigConstants.java

```diff
@@ -191,6 +191,12 @@ public enum EFORegistrationType {
     public static final String REGISTER_STREAM_BACKOFF_EXPONENTIAL_CONSTANT =
             "flink.stream.registerstreamconsumer.backoff.expconst";
 
+    /** The user-provided list of exceptions to recover from. These exceptions are retried indefinitely.
+     * Must be a valid JSON array.
```

Comment:
Passing JSON directly sounds messy, is this done elsewhere in Flink? Can we pass a csv or model the object structure in the keys instead:
```
flink.shard.consumer.error.recoverable=typeA,typeB
```

Or
```
flink.shard.consumer.error.recoverable[0].exception=typeA
flink.shard.consumer.error.recoverable[1].exception=typeB
```